[BUG] false invalid schema

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Hi,
I have an openapi 3.0.3 document file, which contains references for parts of another file (it contains common stuff)
for example:
$ref: './common-components.yaml#/components/responses/UnauthorizedError'

validators that I found say it is valid, also swagger UI is created well from the two files (the parts from common file are included)

But schemathesis says it is invalid:

when 
my_stuff = schemathesis.from_file("./my-file.yaml")
is called, the following error is given:

```
.venv\Lib\site-packages\schemathesis\specs\openapi\loaders.py:256: in from_file
    return from_dict(
.venv\Lib\site-packages\schemathesis\specs\openapi\loaders.py:309: in from_dict
    raise SchemaError(SchemaErrorType.OPEN_API_INVALID_SCHEMA, SCHEMA_INVALID_ERROR)

E   schemathesis.exceptions.SchemaError: The provided API schema does not appear to be a valid OpenAPI schema

```

### To Reproduce

will anonymize, and send example if this is not enough

### Environment

```
- OS: [Windows]
- Python version: [3.11]
- Schemathesis version: [e.g. 3.25.4]
- Spec version: [e.g. Open API 3.0.3]
```

BR,
George

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -9,6 +9,10 @@ Changelog
 - Support for per-media type data generators. :issue:`962`
 - Support for ``application/yaml`` & ``text/yml`` media types in ``YAMLSerializer``.
 
+**Fixed**
+
+- Not resolving references in nested security schemes. :issue:`2073`
+
 **Changed**
 
 - Improve error message when the minimum possible example is too large.

--- a/src/schemathesis/specs/openapi/security.py
+++ b/src/schemathesis/specs/openapi/security.py
@@ -127,7 +127,10 @@ class OpenAPISecurityProcessor(BaseSecurityProcessor):
         security_schemes = components.get("securitySchemes", {})
         if "$ref" in security_schemes:
             return resolver.resolve(security_schemes["$ref"])[1]
-        return security_schemes
+        return {
+            key: resolver.resolve(value["$ref"])[1] if "$ref" in value else value
+            for key, value in security_schemes.items()
+        }
 
     def _make_http_auth_parameter(self, definition: dict[str, Any]) -> dict[str, Any]:
         schema = make_auth_header_schema(definition)

--- a/test/specs/openapi/test_security.py
+++ b/test/specs/openapi/test_security.py
@@ -19,6 +19,22 @@ def test_ref_resolving():
     assert OpenAPISecurityProcessor().get_security_definitions(schema, resolver) == http_schema
 
 
+def test_ref_resolving_nested():
+    http_schema = {"type": "http", "scheme": "basic"}
+    schema = {
+        "openapi": "3.0.0",
+        "info": {"title": "Blank API", "version": "1.0"},
+        "servers": [{"url": "http://localhost/api"}],
+        "paths": {"foo": {"get": {"responses": {"200": {"description": "OK"}}}}},
+        "components": {
+            "securitySchemes": {"basic_auth": {"$ref": "#/components/HTTPSchema"}},
+            "HTTPSchema": http_schema,
+        },
+    }
+    resolver = InliningResolver("", schema)
+    assert OpenAPISecurityProcessor().get_security_definitions(schema, resolver) == {"basic_auth": http_schema}
+
+
 PARAMETER_NAME = "TestApiKey"
 SCHEMA_WITH_PARAMETER_AND_SECURITY_SCHEME = {
     "openapi": "3.0.2",

